internal/host/monitor.go

```diff
@@ -109,6 +109,31 @@ func SortHosts(hosts []*models.Host) ([]*models.Host, bool) {
 	return hosts, allHostsHasInventory
 }
 
+func (m *Manager) resetRoleAssignmentIfNotAllRolesAreSet(cluster *common.Cluster, sortedhotList []*models.Host, tx *gorm.DB) error {
+	var nonAssigned []string
+
+	reply := tx.Model(&models.Host{}).Where("cluster_id = ? and role = ? and (suggested_role = ? or suggested_role = '' or suggested_role is null)",
+		cluster.ID, models.HostRoleAutoAssign, models.HostRoleAutoAssign).Pluck("id", &nonAssigned)
```

Comment:
It will but it will fail if the transaction is SELECT FOR UPDATE from the inventory. We had a bug about it in the past and this is why I am using this idiom

Reply:
OK